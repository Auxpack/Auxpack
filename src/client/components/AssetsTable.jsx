import React from 'react';
import PropTypes from 'prop-types';
// import { makeStyles } from '@material-ui/core/styles';
import { FixedSizeList as List } from 'react-window';


const AssetsTable = props => {
    const assetsArr = (props.build[0].assets.length !== 0) ? props.build[0].assets : [];

    const { getBytes } = props;
    const assetListItems = assetsArr.map((obj, i) => {

        return (<tr key={i}>
            <td>{obj.name}</td>
            <td>{obj.chunks}</td>
            <td>{getBytes(obj.size)}</td>
        </tr>)
    })
    const AssetsCard = () => {
        return (<div className="card large-card darken-1">

            <table className="highlight centered">
                <thead>
                    <tr className="card-body">
                        <th>File Name</th>
                        <th>Chunks</th>
                        <th>File Size</th>
                    </tr>
                </thead>
                <tbody>
                    {assetListItems}
                </tbody>
            </table >
            {/* <ul className="changes-list">
                    {assetListItems}
                </ul> */}

        </div>)
    }

    return <div className="module-container">

        <AssetsCard />

    </div>
}

export default AssetsTable;