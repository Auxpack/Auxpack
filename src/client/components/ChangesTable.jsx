import React from 'react';
import PropTypes from 'prop-types';
import Table from '@material-ui/core/Table';
import { makeStyles } from '@material-ui/core/styles';

const ChangesTable = props => {
    return <div>
        <h3>Changes from Previous Build</h3>
    </div>
}

export default ChangesTable;