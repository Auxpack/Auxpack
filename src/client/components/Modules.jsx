import React from 'react';

const Modules = props => {
    return <div>Modules</div>
}

export default Modules;