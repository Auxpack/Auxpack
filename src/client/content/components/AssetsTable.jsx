import React from 'react';

const AssetsTable = props => {
    const assetsArr = (props.build[0].assets.length !== 0) ? props.build[0].assets : [];

    const { getBytes } = props;
    const assetListItems = assetsArr.map((obj, i) => {

        return (<tr key={i} className="table-row">
            <td>{obj.name}</td>
            <td>{obj.chunks}</td>
            <td>{getBytes(obj.size)}</td>
        </tr>)
    })

    return (<table className="highlight">
        <thead>
            <tr>
                <th>Name</th>
                <th>Chunks</th>
                <th>File Size</th>
            </tr>
        </thead>
        <tbody>
            {assetListItems}
        </tbody>
    </table>)
}


export default AssetsTable;