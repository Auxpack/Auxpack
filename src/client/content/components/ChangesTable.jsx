import React from 'react';
import { makeStyles } from '@material-ui/core/styles';
import ExpansionPanel from '@material-ui/core/ExpansionPanel';
import ExpansionPanelSummary from '@material-ui/core/ExpansionPanelSummary';
import ExpansionPanelDetails from '@material-ui/core/ExpansionPanelDetails';
import Typography from '@material-ui/core/Typography';
import ExpandMoreIcon from '@material-ui/icons/ExpandMore';

const useStyles = makeStyles(theme => ({
    root: {
        width: '100%',
    },
    heading: {
        fontSize: theme.typography.pxToRem(15),
        fontWeight: theme.typography.fontWeightRegular,
    },
}));

const ChangesTable = props => {
    const { dirFinalArrayPrev, dirFinalArray, getBytes } = props;
    // console.log(`prev: `, props.dirFinalArrPrev, `current: `, dirFinalArray)

    // Changes filtering
    const dirFinalFiles = [];
    for (let i = 0; i < dirFinalArray.length; i++) {
        for (let j = 1; j < dirFinalArray[i].length; j++) {
            for (let k = 0; k < dirFinalArray[i][j].length; k++) {
                dirFinalFiles.push([dirFinalArray[i][0] + '/' + dirFinalArray[i][j][k].filename, dirFinalArray[i][j][k].size])
            }
        }
    }

    const dirFinalFilesPrev = [];
    for (let i = 0; i < dirFinalArrayPrev.length; i++) {
        for (let j = 1; j < dirFinalArrayPrev[i].length; j++) {
            for (let k = 0; k < dirFinalArrayPrev[i][j].length; k++) {
                dirFinalFilesPrev.push([dirFinalArrayPrev[i][0] + '/' + dirFinalArrayPrev[i][j][k].filename, dirFinalArrayPrev[i][j][k].size])
            }
        }
    }

    // added files
    const added = dirFinalFiles.filter((curr) => {
        let previous = dirFinalFilesPrev.map((item) => item[0])
        return !previous.includes(curr[0])
    })
    // removed files
    const removed = dirFinalFilesPrev.filter((curr) => {
        let original = dirFinalFiles.map((item) => item[0])
        return !original.includes(curr[0])
    });
    const additions = [];
    if (added.length === 0) {
        additions.push({ path: 'No additions', size: 0 })
    } else {
        for (let i = 0; i < added.length; i += 1) {
            const path = added[i][0]
            const size = added[i][1]
            additions.push({ path, size });
        }
    }
    const removals = [];
    if (removed.length === 0) {
        removals.push({ path: 'No removals', size: 0 })
    } else {
        for (let i = 0; i < removed.length; i += 1) {
            const path = removed[i][0]
            const size = removed[i][1]
            removals.push({ path, size });
        }
    }

    const additionListItems = additions.map((obj, i) => {
        return (<tr key={i} className="table-row">
            <td>{obj.path}</td>
            <td>{getBytes(obj.size)}</td>
        </tr>)
    })
    const removalListItems = additions.map((obj, i) => {
        return (<tr key={i} className="table-row">
            <td>{obj.path}</td>
            <td>{getBytes(obj.size)}</td>
        </tr>)
    })

    const AdditionCard = () => {
        return (<table className="highlight">
            <thead>
                <tr className="card-body">
                    <th>File Path</th>
                    <th>File Size</th>
                </tr>
            </thead>
            <tbody>
                {additionListItems}
            </tbody>
        </table >)

    }

    const RemovalCard = () => {
        return (
            <table className="highlight">
                <thead>
                    <tr className="card-body">
                        <th>File Path</th>
                        <th>File Size</th>
                    </tr>
                </thead>
                <tbody>
                    {removalListItems}
                </tbody>
            </table >

        )
    }

    const SimpleExpansionPanel = () => {
        const classes = useStyles();

        return (
            <div className={classes.root} >
                <ExpansionPanel className="expansionPanel">
                    <ExpansionPanelSummary
                        expandIcon={<ExpandMoreIcon />}
                        aria-controls="panel1a-content"
                        id="panel1a-header"
                    >
                        <Typography className={classes.heading} className="expansion-heading">
                            {/* Expansion heading */}
                            <strong className="centered">Additions</strong>
                        </Typography>
                    </ExpansionPanelSummary>
                    <ExpansionPanelDetails className="panelDetails">
                        {/* Additions Card Panel - content*/}
                        <AdditionCard />
                    </ExpansionPanelDetails>
                </ExpansionPanel>
                {/* Second expansion */}
                <ExpansionPanel className="expansionPanel">
                    <ExpansionPanelSummary
                        expandIcon={<ExpandMoreIcon />}
                        aria-controls="panel1a-content"
                        id="panel1a-header"

                    >
                        {/* Expansion heading */}
                        <Typography className={classes.heading} className="center-heading" className="expansion-heading">
                            <strong className="centered">Removals</strong>
                        </Typography>
                    </ExpansionPanelSummary>
                    <ExpansionPanelDetails className="panelDetails">
                        {/* Removals Card Panel - content*/}
                        <RemovalCard />
                    </ExpansionPanelDetails>
                </ExpansionPanel>

            </div>
        );
    }

    return <div className="cards-container centered">
        <SimpleExpansionPanel />
    </div>
}

export default ChangesTable;
