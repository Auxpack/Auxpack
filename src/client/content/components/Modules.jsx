import React from 'react';
import { makeStyles } from '@material-ui/core/styles';
import ExpansionPanel from '@material-ui/core/ExpansionPanel';
import ExpansionPanelSummary from '@material-ui/core/ExpansionPanelSummary';
import ExpansionPanelDetails from '@material-ui/core/ExpansionPanelDetails';
import Typography from '@material-ui/core/Typography';
import ExpandMoreIcon from '@material-ui/icons/ExpandMore';

const useStyles = makeStyles(theme => ({
    root: {
        width: '100%',
    },
    heading: {
        fontSize: theme.typography.pxToRem(15),
        fontWeight: theme.typography.fontWeightRegular,
    },
}));

const Modules = props => {
    const { dirFinalArray, getBytes } = props;

    const fileRows = dirFinalArray.map((directory) => {
        return directory[1].map((file, j) => (<tr key={file.filename + file.size + j}>
            <td>{file.filename}</td>
            <td>{getBytes(file.size)}</td>
            <td>{file.percentage}</td>
        </tr>)
        )
    })

    const FileTable = () => {
        return (<table className="highlight">
            <thead>
                <tr className="card-body">
                    <th>File Name</th>
                    <th>Size</th>
                    <th>Percentage</th>
                </tr>
            </thead>
            <tbody>
                {fileRows}
            </tbody>
        </table>)
    }
    // const FileTable = props.dirFinalArray.map((directory) => {
    //     const fileListItems = directory[1].map((file, j) => (
    //         <tr key={file.filename + file.size + j}>
    //             <td>{file.filename}</td>
    //             <td>{props.getBytes(file.size)}</td>
    //             <td>{file.percentage}</td>
    //         </tr>
    //     ));

    // return (
    //     <div className="card module-card mw-40 darken-1" key={directory[0]}>
    //         <div className="card-content">
    //             <span className="card-title">{directory[0]}</span>
    //             <table className="highlight centered">
    //                 <thead>
    //                     <tr className="card-body">
    //                         <th>File Name</th>
    //                         <th>File Size</th>
    //                         <th>Percentage</th>
    //                     </tr>
    //                 </thead>
    //                 <tbody>
    //                     {fileListItems}
    //                 </tbody>
    //             </table >
    //         </div>
    //     </div>
    // );
    //     return (<table className="highlight">
    //         <thead>
    //             <tr className="card-body">
    //                 <th>File Name</th>
    //                 <th>File Size</th>
    //                 <th>Percentage</th>
    //             </tr>
    //         </thead>
    //         <tbody>
    //             {fileListItems}
    //         </tbody>
    //     </table >)
    // }); // end fileTable component

    const SimpleExpansionPanel = () => {
        const classes = useStyles();

        return (
            <div className={classes.root} >
                <ExpansionPanel className="expansionPanel" style={{
                    width: '800px',
                }} >
                    <ExpansionPanelSummary
                        expandIcon={<ExpandMoreIcon />}
                        aria-controls="panel1a-content"
                        id="panel1a-header"
                    >
                        <Typography className={classes.heading} className="expansion-heading">
                            {/* Expansion heading */}
                            <strong className="centered">Modules</strong>
                        </Typography>
                    </ExpansionPanelSummary>
                    <ExpansionPanelDetails style={{
                        maxHeight: '60%',
                        overflowY: 'auto'
                    }}>
                        {/* FileTable - content*/}
                        <FileTable />
                    </ExpansionPanelDetails>
                </ExpansionPanel>
            </div>
        );
    }

    return (
        <div className="cards-container centered">
            <SimpleExpansionPanel />
        </div>
    );
}

export default Modules;