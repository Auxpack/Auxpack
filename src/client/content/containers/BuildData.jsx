import React from 'react'

const BuildData = ({ build }) => {

    console.log(`build: `, build)
    return (
        <h2>Build Data</h2>
    );
}

export default BuildData;