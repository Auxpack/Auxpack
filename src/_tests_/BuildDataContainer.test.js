import BuildData from '../client/content/containers/BuildData.jsx';
import ChangesTable from '../client/content/components/ChangesTable.jsx';
import AssetsTable from '../client/content/components/AssetsTable.jsx';
import ErrorsTable from '../client/content/components/Errors.jsx';
import Modules from '../client/content/components/Modules.jsx';
import Tab from '@material-ui/core/Tab';

describe('BuildData Unit Tests', () => {
  
  let shallow = createShallow();
  const props = {
    build:[{
      timeStamp:1575426090404,
      time:1439,
      hash:"546142ce1b49a6ba7d6f",
      errors:[],
      size:1172113,
      assets:[{"name":"bundle.js","chunks":["main"],"size":1172113}],
      chunks:[{"size":1118609,"files":["bundle.js"],"modules":[{"name":"./client/App.jsx","size":6375,"id":"./client/App.jsx"}]}],
      treeStats:{
        "cjs":[{"name":"./test2.js","size":49,"reasonTypes":[{"name":"./test.js","type":"cjs require"}]},{"name":"./test2.js","size":49,"reasonTypes":[{"name":"./test.js","type":"cjs require"}]}],
        "esm":[{"name":"./client/App.jsx","size":6375,"reasonTypes":[{"name":"./client/index.js","type":"harmony side effect evaluation"}]}],
        "both":[{"name":"./node_modules/react/index.js","size":190,"reasonTypes":[{"name":"./client/App.jsx","type":"harmony side effect evaluation"}]}]}
      }],
    activeBuild: 0,
    getBytes: function() {return},
    dirFinalArray: [],
    dirFinalArrayPrev: []
  }

  describe('BuildData Container', () => {
    let wrapper;

    beforeAll(() => {
      wrapper = shallow(<BuildData {...props} />)
    });

    it('BuildData snapshot testing', () => {
      expect(wrapper).toMatchSnapshot();
    });

    it('BuildData should display four Tab components', () => {
      expect((wrapper.find(Tab)).length).toBe(4);
    });

    it('BuildData should render ChangesTable, AssetsTable, ErrorsTable, and Modules components', () => {
      expect((wrapper.find(ChangesTable)).length).toBe(1);
      expect((wrapper.find(AssetsTable).length)).toBe(1);
      expect((wrapper.find(ErrorsTable)).length).toBe(1);
      expect((wrapper.find(Modules)).length).toBe(1);
    });
  });

  describe('ChangesTable', () => {
    let wrapper;

    beforeAll(() => {
      wrapper = shallow(<ChangesTable {...props} />)
    });

    it('ChangesTable snapshot testing', () => {
      expect(wrapper).toMatchSnapshot();
    });

    it('ChangesTable should render div with class "cards-container" wrapped around an Expansion Panel', () => {
      expect((wrapper.find('div')).hasClass('cards-container')).toBe(true);
      expect((wrapper.find('SimpleExpansionPanel')).length).toBe(1);
    });
  });

  describe('AssetsTable', () => {
    let wrapper;

    beforeAll(() => {
      wrapper = shallow(<AssetsTable {...props} />)
    });

    it('AssetsTable snapshot testing', () => {
      expect(wrapper).toMatchSnapshot();
    });

    it('AssetsTable should render table', () => {
      expect((wrapper.find('table')).length).toBe(1);
    });
  });

  describe('ErrorsTable', () => {
    let wrapper;

    beforeAll(() => {
      wrapper = shallow(<ErrorsTable {...props} />)
    });

    it('ErrorsTable snapshot testing', () => {
      expect(wrapper).toMatchSnapshot();
    });

    it('ErrorsTable should render table', () => {
      expect((wrapper.find('table')).length).toBe(1);
    });
  });

  describe('Modules', () => {
    let wrapper;

    beforeAll(() => {
      wrapper = shallow(<Modules {...props} />)
    });

    it('Modules snapshot testing', () => {
      expect(wrapper).toMatchSnapshot();
    });

    it('Modules should render div with class "cards-container" wrapped around an Expansion Panel', () => {
      expect((wrapper.find('div')).hasClass('cards-container')).toBe(true);
      expect((wrapper.find('SimpleExpansionPanel')).length).toBe(1);
    });
  });
})


