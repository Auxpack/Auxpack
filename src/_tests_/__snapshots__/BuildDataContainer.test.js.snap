// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BuildData Unit Tests AssetsTable AssetsTable snapshot testing 1`] = `ShallowWrapper {}`;

exports[`BuildData Unit Tests AssetsTable ChangesTable snapshot testing 1`] = `ShallowWrapper {}`;

exports[`BuildData Unit Tests BuildData BuildData snapshot testing 1`] = `ShallowWrapper {}`;

exports[`BuildData Unit Tests BuildData Container BuildData snapshot testing 1`] = `ShallowWrapper {}`;

exports[`BuildData Unit Tests ChangesTable ChangesTable snapshot testing 1`] = `ShallowWrapper {}`;

exports[`BuildData Unit Tests ErrorsTable ChangesTable snapshot testing 1`] = `ShallowWrapper {}`;

exports[`BuildData Unit Tests ErrorsTable ErrorsTable snapshot testing 1`] = `ShallowWrapper {}`;

exports[`BuildData Unit Tests ErrorsTable ErrorsTable snapshot testing 2`] = `ShallowWrapper {}`;

exports[`BuildData Unit Tests Modules ErrorsTable snapshot testing 1`] = `ShallowWrapper {}`;

exports[`BuildData Unit Tests Modules Modules snapshot testing 1`] = `ShallowWrapper {}`;
